demos/dev_mode_key_provisioning/src/aws_key_provisioning_demo.c

```diff
@@ -0,0 +1,613 @@
+        pucSigPtr += 2;
+
+        memcpy( pucSigPtr, &ucTemp[ 32 ], 32 );
+    }
+    else
+    {
+        pucSigPtr[ 0 ] = 0x20;
+        pucSigPtr++;
+        memcpy( pucSigPtr, &ucTemp[ 32 ], 32 );
+    }
+
+    *pxSigLen = ( CK_ULONG ) pucSig[ 1 ] + 2;
+
+    if( xResult != 0 )
+    {
+        configPRINTF( ( "ERROR: Failure in signing callback: %d \r\n", xResult ) );
+        lFinalResult = TLS_ERROR_SIGN;
+    }
+
+    return lFinalResult;
+}
+
+/*-----------------------------------------------------------*/
+
+/* @brief Provision Device Using Device and CA Certificates
+ *
+ *
+ *	\return    Void
+ *  Prints error message upon failure, success message upon success
+ *
+ */
+static void vProvisionDevice( void )
+{
+    /* Provisioning Device Certificate */
+    CK_OBJECT_HANDLE xObject = 0;
+
+    xResult = xProvisionCertificate( xGlobalSession,
+                                     keyCLIENT_CERTIFICATE_PEM,
+                                     sizeof( keyCLIENT_CERTIFICATE_PEM ),
+                                     ( uint8_t * ) pkcs11configLABEL_DEVICE_CERTIFICATE_FOR_TLS,
+                                     &xObject );
+
+    if( ( xResult != CKR_OK ) || ( xObject == CK_INVALID_HANDLE ) )
+    {
+        configPRINTF( ( "ERROR: Failed to provision device certificate. %d \r\n", xResult ) );
+    }
+
+    /* Provisioning JITR CA Certificate */
+    if( xResult == CKR_OK )
+    {
+        if( sizeof( keyJITR_DEVICE_CERTIFICATE_AUTHORITY_PEM ) != 0 )
+        {
+            xResult = xProvisionCertificate( xGlobalSession,
+                                             keyJITR_DEVICE_CERTIFICATE_AUTHORITY_PEM,
+                                             sizeof( keyJITR_DEVICE_CERTIFICATE_AUTHORITY_PEM ),
+                                             ( uint8_t * ) pkcs11configLABEL_JITP_CERTIFICATE,
+                                             &xObject );
+
+            xResult = CKR_OK;
+        }
+    }
+
+    if( xResult == CKR_OK )
+    {
+        configPRINTF( ( "Device credential provisioning succeeded.\r\n" ) );
+    }
+    else
+    {
+        configPRINTF( ( "Device credential provisioning failed.\r\n" ) );
+    }
+}
+
+/*-----------------------------------------------------------*/
+
+/* @brief Provisions a Device Using Internally Generated Keys
+ *
+ *
+ *  \return Void function, console outputs upon success/failure
+ *
+ */
+static void vDevModeDeviceKeyProvisioning( void )
+{
+    xResult = xInitializePkcs11Session( &xGlobalSession );
+
+    if( xResult != CKR_OK )
+    {
+        configPRINTF( ( "Failed to open PKCS #11 session.\r\n" ) );
+    }
+
+    xResult = C_GetFunctionList( &pxGlobalFunctionList );
+
+    if( xResult != CKR_OK )
+    {
+        configPRINTF( ( "Failed to get function list.\r\n" ) );
+    }
+
+    #if ( !DEMO_PART )
+        {
+            CredentialsProvisioned_t xCurrentCredentials = eStateUnknown;
+
+            /* Note that mbedTLS does not permit a signature on all 0's. */
```

Comment:
Technically, I think the issue is what happens when performing the resulting exponentiation in ECDSA - so the problem is the math, not the library.

Reply:
I copied this comment from a test file, but removed it to avoid any confusion. Thanks.